base_rest/models/rest_service_registration.py

```diff
@@ -329,7 +329,14 @@ def _generate_methods(self):
                 default_route = routes[0]
                 rule = Rule(default_route)
                 Map(rules=[rule])
-                if rule.arguments:
+                if rule.arguments and routing["params_as_kwargs"]:
+                    method = METHOD_TMPL_WITH_KWARGS.format(
```

Comment:
Thank you @etobella. 
 As stated by the werkzeug documentation `besides the rule-string all arguments *must* be keyword arguments  in order to not break the application on Werkzeug upgrades` https://github.com/pallets/werkzeug/blob/master/src/werkzeug/routing.py#L548

The problem here is to be able to dissociate path parameters (validated by werkzeuk) from others parameters (query / json) validated by the rest framework. In the current implementation I expect that path parameters are always pass to the final method into the same order as the one defined into the url. Unfortunately it's seems it's not always the case when werkzeug calls the generated method.

IMO it would be safe to be able to not break the convention that path parameters are *args in the same order that into the path definition and **params are query/json paramters. We could maybe achieve this by only modifying the generated method....


```python
METHOD_TMPL_WITH_ARGS = """
def {method_name}(self, {path_parameters}, **kwargs):
    return self._process_method(
        "{service_name}",
        "{service_method_name}",
        *[{args}],
        params=kwargs
    )
"""

if rule.arguments:
    method = METHOD_TMPL_WITH_ARGS.format(
        method_name=method_name,
        service_name=self._service_name,
        service_method_name=name,   
        path_parameters = "," . join(["%s=None" for arg in rule.arguments]),
        args=", ".join(rule.arguments),
    )
    

```

What do you thingk about?

Reply:
The problem I found is that in some cases (on my case, if the a rest call is sent before the modules are fully loaded), werkzeug is passing the parameters in the inverse order, so, when generating the method, the passed parameters will not preserve the order.
For example, If I have the
`/<string:param1>/<int:param2>`
the order is important, and it might happen two options:
```
def my_method(self, param1=None, param2=None, **kwargs):
    return self._process_method(
        "my_method",
        "{service_method_name}",
        *[param1, param2],
        params=kwargs
    )
```
or
```
def my_method(self, param2=None, param1=None, **kwargs):
    return self._process_method(
        "my_method",
        "{service_method_name}",
        *[param2, param1],
        params=kwargs
    )
```
As you can see, the results will be different

Reply:
@etobella  In order to preserve the order in every cases we could extract the parameters list into the right order from `rule.string` https://github.com/pallets/werkzeug/blob/master/src/werkzeug/routing.py#L551

Reply:
I think that making this is even more complex, as we are trying to fix something because we are not using the expected behavior

> As stated by the werkzeug documentation besides the rule-string all arguments *must* be keyword arguments in order to not break the application on Werkzeug upgrades https://github.com/pallets/werkzeug/blob/master/src/werkzeug/routing.py#L548